src/python/include/Alpha_complex_interface.h

```diff
@@ -23,45 +23,74 @@
 #include <iostream>
 #include <vector>
 #include <string>
+#include <memory>  // for std::unique_ptr
 
 namespace Gudhi {
 
 namespace alpha_complex {
 
 class Alpha_complex_interface {
-  using Dynamic_kernel = CGAL::Epeck_d< CGAL::Dynamic_dimension_tag >;
-  using Point_d = Dynamic_kernel::Point_d;
+ private:
+  using Exact_kernel = CGAL::Epeck_d<CGAL::Dynamic_dimension_tag>;
+  using Inexact_kernel = CGAL::Epick_d<CGAL::Dynamic_dimension_tag>;
+  using Point_exact_kernel = typename Exact_kernel::Point_d;
+  using Point_inexact_kernel = typename Inexact_kernel::Point_d;
 
- public:
-  Alpha_complex_interface(const std::vector<std::vector<double>>& points) {
-    auto mkpt = [](std::vector<double> const& vec){
-      return Point_d(vec.size(), vec.begin(), vec.end());
-    };
-    alpha_complex_ = new Alpha_complex<Dynamic_kernel>(boost::adaptors::transform(points, mkpt));
+  template <typename CgalPointType>
+  std::vector<double> pt_cgal_to_cython(CgalPointType& ph) {
+    std::vector<double> vd;
+    for (auto coord = ph.cartesian_begin(); coord != ph.cartesian_end(); coord++)
+      vd.push_back(CGAL::to_double(*coord));
```

Comment:
I wonder if we should call CGAL::exact(ph) at the beginning of this function, in the 'exact' case.
What is the 'h' for in 'ph'?

Reply:
I renamed ph as point on e7ccfabc
How could I check if I need to call CGAL::exact(point) ?

Reply:
Ah, this function is only ever used for the input points, so there is no need to call CGAL::exact.